.github/workflows/host-tools.yml

```diff
@@ -0,0 +1,111 @@
+            pageNum=$((pageNum + 1))
+            if [ $(echo $json | jq length) -eq 0 ]; then
+              # This means that we've passed the end and reached an empty array
+              break
+            fi
+            # Store array of revisions
+            revisions=($(echo $json | jq -e -r ".[] | .html_url | select(contains(\"hosttools\"))[${#releaseUrl}:]")) \
+              || continue
+            hrev=${revisions[0]}
+          done
+          echo "hrev=$hrev" >> "$GITHUB_OUTPUT"
+      - name: Build host tools
+        if: steps.get_hrev.outputs.hrev != steps.get_hosttools.outputs.hrev
+        run: |
+          echo "Building host tools at ${{ steps.get_hrev.outputs.hrev }}"
+          hosttools="
+            <build>addattr
+            <build>anyboot
+            <build>bfs_shell
+            <build>collectcatkeys
+            <build>copyattr
+            <build>create_image
+            <build>create_repository_config
+            <build>data_to_source
+            <build>fat_shell
+            <build>fixup_amiga_boot_checksum
+            <build>fixup_tos_boot_checksum
+            <build>fs_shell_command
+            <build>generate_attribute_stores
+            <build>get_package_dependencies
+            <build>linkcatkeys
+            <build>makebootable
+            <build>mbrtool
+            <build>mimeset
+            <build>package
+            <build>package_repo
+            <build>rc
+            <build>resattr
+            <build>rm_attrs
+            <build>set_haiku_revision
+            <build>settype
+            <build>setversion
+            <build>unzip
+            <build>update_package_requires
+            <build>vmdkimage
+            <build>xres
+          "
+          git clone https://review.haiku-os.org/haiku --depth 1
+          git clone https://review.haiku-os.org/buildtools --depth 1
+          sudo apt install -y git nasm bc autoconf automake texinfo flex bison \
+            gawk build-essential unzip wget zip less zlib1g-dev libzstd-dev \
+            xorriso libtool python3 attr
+          cd buildtools/jam
+          make -j$(nproc)
+          cd $GITHUB_WORKSPACE/haiku
+          mkdir generated.${{ env.HAIKU_ARCH }}; cd generated.${{ env.HAIKU_ARCH }}
+          ../configure --cross-tools-source ../../buildtools --build-cross-tools ${{ env.HAIKU_ARCH }} -j$(nproc)
+          $GITHUB_WORKSPACE/buildtools/jam/jam0 -j$(nproc) -q $hosttools
+          zip -r -j x86_64-linux-hosttools-${{ steps.get_hrev.outputs.hrev }}.zip \
+            objects/linux/lib \
+            objects/linux/x86_64/release/tools \
+            ../License.md \
+            -x *.o
+      - name: Upload artifacts
+        if: steps.get_hrev.outputs.hrev != steps.get_hosttools.outputs.hrev
+        uses: actions/upload-artifact@v3
+        with:
+          name: x86_64-linux-hosttools-${{ steps.get_hrev.outputs.hrev }}
+          path: x86_64-linux-hosttools-${{ steps.get_hrev.outputs.hrev }}.zip
```

Comment:
You zipped the file while you were in `$GITHUB_WORKSPACE/haiku`. This means the path should be `haiku/[name]`. Alternatively, it's possible to use an absolute path for the output of the `zip` command above.

Currently, nothing is being uploaded due to the package file is missing.

Reply:
Oops, should be fine now, using same template as package.yml.